feature/staticroute/otg_tests/basic_static_route_support_test/basic_static_route_support_test.go

```diff
@@ -343,14 +343,21 @@ func (td *testData) testRecursiveNextHopResolutionDisabled(t *testing.T) {
 
 	t.Run("Telemetry", func(t *testing.T) {
 
-		gnmi.Await(t, td.dut, sp.Static(td.staticIPv4.cidr(t)).Prefix().State(), 30*time.Second, td.staticIPv4.cidr(t))
-		gnmi.Await(t, td.dut, sp.Static(td.staticIPv6.cidr(t)).Prefix().State(), 30*time.Second, td.staticIPv6.cidr(t))
 		// Validate static route next-hop recursive lookup is disabled
-		if got, want := gnmi.Get(t, td.dut, sp.Static(td.staticIPv4.cidr(t)).NextHop("0").Recurse().State()), false; got != want {
-			t.Errorf("IPv4 Static Route next hop: got: %v, want: %v", got, want)
+		_, v4RecurseSet := gnmi.Watch(t, td.dut, sp.Static(td.staticIPv4.cidr(t)).State(), 30*time.Second, func(v *ygnmi.Value[*oc.NetworkInstance_Protocol_Static]) bool {
+			val, present := v.Val()
+			return !present || (present && !val.GetNextHop("0").GetRecurse())
+		}).Await(t)
+		if !v4RecurseSet {
+			t.Log("Unable to set recurse to false for v4 prefix")
```

Comment:
seems like this should be `t.Errrorf`? if the recurse is not working as expected, why is it downgraded to a log message?

Reply:
There are more changes to file. This is modified to error in latest file.